cli/edp/cli.py

```diff
@@ -114,25 +112,75 @@ def _ingest(project, cycle, api_url, api_key, dir):
             cell_id = row['item_id']
             schedule_file = row['schedule_file_name']
 
-        click.echo('Uploading meta data file')
+        click.echo(click.style('Uploading meta data file', fg='blue'))
         meta_file = gc.uploadFileToFolder(data_folder['_id'], meta_file)
-        click.echo('Uploading data file')
-        data_file = gc.uploadFileToFolder(data_folder['_id'], data_file)
+        click.echo(click.style('Uploading data file', fg='blue'))
+        data_file = gc.uploadFileToFolder(data_folder['_id'], data_file_path)
+
+        if summary_func is not None:
+            print(data_file_path)
+            summary = summary_func(data_file_path)
 
         test = {
+            'name': test_name,
             'startDate': start_date,
             'cellId': cell_id,
             'batteryType': '',
             'channel': channel,
             'scheduleFile': schedule_file,
             'metaDataFileId': meta_file['_id'],
-            'dataFileId': data_file['_id']
+            'dataFileId': data_file['_id'],
+            'public': 'public',
+            'summary': summary
         }
 
         test = gc.post('edp/projects/%s/cycles/%s/batches/%s/tests' % (project, cycle, batch['_id']), json=test)
 
-        click.echo(click.style('Test created: %s' % test['_id'], fg='green'))
+        click.echo(click.style('Test created: %s' % test['_id'], fg='blue'))
+
+@cli.command('ingest', help='Ingest data')
+@click.option('-p', '--project', default=None, help='the project id', required=True)
+@click.option('-c', '--cycle', default=None, help='the cycle id', required=True)
+@click.option('-d', '--dir', help='path to batch(es) to ingest',
+              type=click.Path(exists=True, dir_okay=True, file_okay=False, readable=True), default='.')
+@click.option('-u', '--api-url', default='http://localhost:8080/api/v1', help='RESTful API URL '
+                   '(e.g https://girder.example.com/api/v1)')
+@click.option('-k', '--api-key', envvar='GIRDER_API_KEY', default=None,
+              help='[default: GIRDER_API_KEY env. variable]', required=True)
+@click.option('-b', '--public', is_flag=True,
+              help='Marked the data as public')
+@click.option('-s', '--summary-func', default=None, help='Fully qualified name of function to summarize test data.')
+def _ingest(project, cycle, api_url, api_key, dir, public, summary_func):
+    gc = GC(api_url=api_url, api_key=api_key)
+
+    # Try to get edp data folder
+    data_folder = gc.resourceLookup('/collection/edp/data', test=True)
+
+    # Create a private folder
+    if data_folder is None:
+
+        me = gc.get('/user/me')
+        private_folder = next(gc.listFolder(me['_id'], 'user', 'Public' if public else 'Private' ))
```

Comment:
Do we need to check for `StopIteration` in case `listFolder` returns and empty list? (i.e.  the user has deleted the `Public` or `Private` folder?)

Reply:
Sure, I can add that